components/camera/client.go

```diff
@@ -228,6 +173,33 @@ func (c *client) Stream(
 	return stream, nil
 }
 
+func (c *client) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, ImageMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::client::Image")
+	defer span.End()
+	expectedType, _ := utils.CheckLazyMIMEType(mimeType)
+
+	convertedExtra, err := goprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+	resp, err := c.client.GetImage(ctx, &pb.GetImageRequest{
```

Comment:
ah one thing - if in here or the server an image is returned as nill we should gracefully handle that.

Otherwise, especially with the server code, since we have a byte array that someone could write a driver against we could panic.


See olivia's old pr for an example of this: https://github.com/viamrobotics/rdk/commit/453778f8b6e785c89fba3be3c879743c54f4475a

Reply:
gotcha, I'll write an error. Thanks for the callout